src/pydata_sphinx_theme/assets/scripts/index.js

```diff
@@ -43,23 +44,19 @@ function scrollToActive() {
   );
 
   if (!isNaN(storedScrollTop)) {
+    // If we've got a saved scroll position, just use that
     sidebar.scrollTop = storedScrollTop;
   } else {
-    var active_pages = sidebar.querySelectorAll(".active");
-    var offset = 0;
-    var i;
-    for (i = active_pages.length - 1; i > 0; i--) {
-      var active_page = active_pages[i];
-      if (active_page !== undefined) {
-        offset += active_page.offsetTop;
+    // Otherwise, calculate a position to scroll to based on the lowest `active` link
+    var active_pages = sidebarNav.querySelectorAll(".active");
+    if (active_pages.length > 0) {
+      // Use the last active page as the offset since it's the page we're on
+      var offset = active_pages[active_pages.length - 1].offsetTop;
```

Comment:
Your memory is correct! I wasn't sure what the benefit was of calculating the offsetTop for all active items - don't we only ever care about the final active item, since that will be the page that we are on? What's the reason to calculate it for all of them?

Reply:
So I think this was done because `offsetTop` is relative to the parent div (https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop). And so because we have a set of nested divs here, if we want to know the actual, total offset of the page link to the top of the sidebar, we need to sum all those offsets (or find a property that gives this total directly).

Exploring this in the console of the browser for one of the sub-sub pages (https://pydata-sphinx-theme--534.org.readthedocs.build/en/534/demo/subpages/subsubpages/subsubpage5.html):


![image](https://user-images.githubusercontent.com/1020496/148953841-c26b32b6-2cb7-4972-96c5-87df7ead7376.png)

Reply:
(this might also be solvable with some `position` mangling which might determine which is the offset parent div .., but not directly understanding this)

Reply:
Ahh good point. I wonder if we could use this to be more efficient in the code:

https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect

Reply:
I think the problem is that that property gives the offset to the "viewport", so distance with the browser visible screen edge, while we need the offset to the actual page, so also what is hidden by scrolling down. 

I would have thought that there was also some property to directly get that, but I didn't find anything back when doing the original PR.

Reply:
When you say "to the actual page", what do you mean? The just need the right information to set `scrollTop` equal to the right value, right? What do you think about the implementation here: https://github.com/pydata/pydata-sphinx-theme/pull/534/commits/8ef7370affae98620627c3527034c71757c4e203

That calculates the `y` value of both the sidebar and the latest active item, and scrolls according to the diff